File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    for s,e in queries:
        temp=list(my_string[s:e+1])
        my_string=my_string[:s]+''.join(c for c in temp[::-1])+my_string[e+1:]
    r=my_string
    return r

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>
using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    string r="";
    int len;
    for(int i=0;i<queries.size();i++) {
        len=queries[i][1]-queries[i][0];
        if(len%2!=0) len+=1;
        for(int j=0;j<len/2;j++) {
            char t=my_string[queries[i][0]+j];
            my_string[queries[i][0]+j]=my_string[queries[i][1]-j];
            my_string[queries[i][1]-j]=t;
        }
    }
    r=my_string;
    return r;
}